src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.cs

```diff
@@ -563,9 +563,19 @@ public ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter)
             BlockParameter fromBlock = filter.FromBlock;
             BlockParameter toBlock = filter.ToBlock;
 
-            return ResultWrapper<IEnumerable<FilterLog>>.Success(_blockchainBridge.GetLogs(fromBlock, toBlock,
-                filter.Address,
-                filter.Topics));
+            try
+            {
+                return ResultWrapper<IEnumerable<FilterLog>>.Success(_blockchainBridge.GetLogs(fromBlock, toBlock, filter.Address, filter.Topics));
+            }
+            catch (ArgumentException e)
+            {
+                switch (e.Message)
+                {
+                    case "Block not found.": return ResultWrapper<IEnumerable<FilterLog>>.Fail(e.Message, ErrorCodes.ResourceNotFound);
```

Comment:
can we then move it to some constant string <- looks like it is used for matching